### PR TITLE
[VideoPlayer] define message queue time size in a single place

### PR DESCRIPTION
**Pull request by thexai**
## Description
- Define message queue time size in a single place.
- Add audio/video queue time to debug OSD.

## Motivation and context
Currently audio/video queue times are hard coded in several places and mixed seconds (8.0) and milliseconds (8000.0).

This PR consolidate to a single definition in one place to easy change later or experiment with different values. 

Changes are based on https://github.com/xbmc/xbmc/pull/25330 and this allows merge some of these changes fast and reduce complexity of the other PR.

@a1rwulf FYI

Some time ago I had also thought about reducing the queue size to 4 seconds because I think 8 seconds is too much for now days conditions with very high UHD bitrates but this is reserved for follow-up PR. This PR not has any functional change except of add a/v queue duration time to debug info OSD.

## How has this been tested?
Runtime Windows x64

## What is the effect on users?
Nothing except some more info in debug OSD.

## Screenshots (if appropriate):

**Before**
before

**After**
after

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by enen92** [comment]
Nice!

File: xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp [hunk 1]
```diff
@@ -67,9 +67,11 @@ CVideoPlayerVideo::CVideoPlayerVideo(CDVDClock* pClock
   m_iDroppedRequest = 0;
   m_fForcedAspectRatio = 0;
 
-  // 128 MB allows max bitrate of 128 Mbit/s (e.g. UHD Blu-Ray) during 8 seconds
-  m_messageQueue.SetMaxDataSize(128 * 1024 * 1024);
-  m_messageQueue.SetMaxTimeSize(8.0);
+  m_messageQueueTimeSize = messageQueueTimeSize;
```
Comment (enen92) [inline xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp:70]: you can probably use the initializer list

File: xbmc/cores/VideoPlayer/VideoPlayer.cpp [hunk 2]
```diff
@@ -636,6 +644,7 @@ CVideoPlayer::CVideoPlayer(IPlayerCallback& callback)
   m_HasVideo = false;
   m_HasAudio = false;
   m_UpdateStreamDetails = false;
+  m_messageQueueTimeSize = VP_MESSAGE_QUEUE_TIME_SIZE;
```
Comment (enen92) [inline xbmc/cores/VideoPlayer/VideoPlayer.cpp:647]: I find the "timesize" a bit confusing despite not being much familiar with the vp timing internals (probably refers to a scale applied to the time size?). Wouldn't it make more sense to use std::chrono for the constexpr and then use the absolute values in seconds/milliseconds/wtv depending on the actual use?

**Review by a1rwulf — APPROVED** [review]
Thanks for splitting out those changes @thexai.
I also agree that 4s should be enough thease days and reduce memory usage on high bitrate files considerably.

File: xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp (hunk 1 above)
Comment (thexai) [reply xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp]: done

**Comment by thexai** [comment]
Changed to use initialization list and adds a missing replacement of other literal value.

**Comment by thexai** [comment]
> I also agree that 4s should be enough thease days and reduce memory usage on high bitrate files considerably.

Yes, reduce memory usage is the main reason: for 128 Mbit/s video stream 8 seconds may suppose up to 128 MB of ram usage only for this, additional to FileCahe (others 64, 128, ... MB) depending of user settings.

Change to 4s reduces this to half 64 MB for typical max 128 Mbit/s video stream. Since default FileCache setting is also 64 MB then is 64 MB + 64 MB with default settings.

Additionally, the variations or micro-interruptions in the local network are on the scale of milliseconds, so it's not necessary store 8 seconds of audio/video demuxed packets in RAM to ensure smooth playback.

**Comment by popcornmix** [comment]
You should read https://github.com/xbmc/xbmc/issues/24745 when considering reducing the message queue size.
There is an issue with VP where "bad stuff" happens when the seek time of audio and video streams exceeds the message queue size duration. This is particularly an issue with ISA where seek offsets may be separated by 10s.

I actually did a similar local change where I increased the message queue size duration to 16s which makes seeking youtube (and similar ISA) videos much more fluent.

File: xbmc/cores/VideoPlayer/VideoPlayer.cpp (hunk 2 above)
Comment (thexai) [reply xbmc/cores/VideoPlayer/VideoPlayer.cpp:647]: > Wouldn't it make more sense to use std::chrono for the constexpr and then use the absolute values in seconds/milliseconds/wtv depending on the actual use?

I'm not sure if worth it: ms seems only used in two places vs 7 or 8 places that wants seconds (double).  While std::chrono is easy to convert to int with `.count()` i'm not sure if to convert to double requires additional `static_cast` anyway.  

Now only rounded values ​​8.0 or 4.0 are used but with the low latency PR they will probably want to try lower values ​​like 0.5s which would be cumbersome (for chrono) if everything is not implemented from the beginning in ms... I see it is more practical for now in seconds and double.

**Pull request merged**